### PR TITLE
Feature SpinUp

### PR DESCRIPTION
**Pull request by niteflyunicorns**
SpinUp feature functionality. 
Addreses #379.

### CODE REVIEW

File: src/SW_Domain.c [hunk 1]
```diff
@@ -156,6 +156,57 @@ void SW_DOM_read(SW_DOMAIN* SW_Domain, LOG_INFO* LogInfo) {
                                 Time_get_lastdoy_y(SW_Domain->endyr) : 365;
                 fenddy = swTRUE;
                 break;
+            case 8: // Spinup Mode
+                y = atoi(value);
+                
+                if (y != 1 || y != 2) {
```
Comment (dschlaep) [inline src/SW_Domain.c:162]: The check against bad user input for spinup mode `y != 1 || y != 2` is always true -- see "warning: overlapping comparisons always evaluate to true [-Wtautological-overlap-compare]"
For example, if y = 1, then y != 2 is true; if y = 2, then y != 1 is true; if y = 3, then both y != 1 and y != 2 are true.

It appears that the desired logic should be `!(y == 1 || y == 2)` or (equivalently) `y != 1 && y != 2`

File: src/SW_Control.c [hunk 2]
```diff
@@ -518,6 +525,86 @@ void SW_CTL_run_current_year(SW_ALL* sw, SW_OUTPUT_POINTERS* SW_OutputPtrs,
...
+          years[ i + ( scope * quotient ) ] = yr + i;
+        }
+      }
+
+      break;
+    default: // same as case 1
+      // initialize random array
+      for ( i = 0; i < duration; i++ ) {
+        yr = RandUniIntRange( sw->Model.startyr,
+                                      finalyr,
```
Comment (dschlaep) [inline src/SW_Control.c:584]: We define `spinup_scope` as "first N years of simulation" ; however, we set `finalyr = sw->Model.startyr + scope` and `RandUniIntRange()` includes the bounds. This results that mode 1 uses the first N + 1 years. Thus, we need `yr = RandUniIntRange( sw->Model.startyr, finalyr - 1, ...)` -- or `finalyr = sw->Model.startyr + scope - 1`.

File: src/SW_Control.c [hunk 3]
```diff
@@ -518,6 +525,92 @@ void SW_CTL_run_current_year(SW_ALL* sw, SW_OUTPUT_POINTERS* SW_OutputPtrs,
...
+          as specified by the given spinup scope and duration which
+          then calls 'SW_SWC_water_flow' for each day.
+
+@param[in,out] sw Comprehensive struct of type SW_ALL containing all
+  information in the simulation
+@param[in,out] SW_OutputPtrs SW_OUTPUT_POINTERS of size SW_OUTNKEYS which
+  hold pointers to subroutines for output keys
+@param[in,out] LogInfo Holds information dealing with logfile output
+*/
+void SW_CTL_run_spinup(SW_ALL* sw, SW_OUTPUT_POINTERS* SW_OutputPtrs,
```
[the remark this answers is not in the document]
Comment (niteflyunicorns) [reply src/SW_Control.c:539]: I agree this is a good course of action. I am including it in my next round of edits.

**Review by dschlaep — COMMENTED** [review]
Looks great, thanks for addressing my comments!

**Pull request merged**